astroid/protocols.py

```diff
@@ -397,24 +397,25 @@ def arguments_assigned_stmts(
     context: InferenceContext | None = None,
     assign_path: list[int] | None = None,
 ) -> Any:
-    if context.callcontext:
+    try:
```

Comment:
`node_classes.AssignedStmtsPossibleNode` (typing of `node`) includes types that do not have a `name` attribute. `mypy` rightfully complains about this on old L417 (and others) as it could create an uncaught `AttributeError`.

I have resolved this by looking up the attribute within this `try ... except`.

Why this isn't an issue is because of the particular way in which these methods work. Apparently we never reach this method with the full spectrum of `node_classes.AssignedStmtsPossibleNode`, but only with a subset of them. However, the order in which these methods are called can easily contain 10 calls so it's quite hard to determine the subset.

I thought it would be better to just be able to pass all nodes as typed instead of narrowing the function down.

_The same is true about `context`. The fact that we never crashed without the `and context` shows that we never actually rely on the default value of `context`, but it is better to make the function work with it._

Reply:
https://github.com/PyCQA/astroid/pull/1644/files#r901602217